napari_tiff/napari_tiff_metadata.py

```diff
@@ -0,0 +1,395 @@
+from typing import Any
+
+import numpy
+from tifffile import PHOTOMETRIC, TiffFile, xml2dict
+from vispy.color import Colormap
+
+from napari_tiff.napari_tiff_colormaps import (
+    alpha_colormap,
+    cmyk_colormaps,
+    int_to_rgba,
+    rgb_colormaps,
+)
+
+
+def get_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return metadata keyword argument dictionary for napari layer."""
+    if tif.is_ome:
+        metadata_kwargs = get_ome_tiff_metadata(tif)
+    # TODO: combine interpretation of imagej and tags metadata?:
+    elif tif.is_imagej:
+        metadata_kwargs = get_imagej_metadata(tif)
+    else:
+        metadata_kwargs = get_tiff_metadata(tif)
+
+    # napari does not use this this extra `metadata` layer attribute
+    # but storing the information on the layer next to the data
+    # will allow users to access it and use it themselves if they wish
+    metadata_kwargs["metadata"] = get_extra_metadata(tif)
+
+    return metadata_kwargs
+
+
+def get_extra_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return any non-empty tif metadata properties as a dictionary."""
+    metadata_dict = {}
+    empty_metadata_values = [None, "", (), []]
+    for name in dir(tif.__class__):
+        obj = getattr(tif.__class__, name)
+        if "metadata" in name:
+            metadata_value = obj.__get__(tif)
+            if metadata_value not in empty_metadata_values:
+                print(metadata_value)
+                if isinstance(metadata_value, str):
+                    try:
+                        metadata_value = xml2dict(metadata_value)
+                    except Exception:
+                        pass
+                metadata_dict[name] = metadata_value
+    return metadata_dict
+
+
+def get_tiff_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return napari metadata from largest image series in TIFF file."""
+    # TODO: fix (u)int32/64
+    # TODO: handle complex
+    series = tif.series[0]
+    for s in tif.series:
+        if s.size > series.size:
+            series = s
+    dtype = series.dtype
+    axes = series.axes
+    shape = series.shape
+    page = next(p for p in series.pages if p is not None)
+    extrasamples = page.extrasamples
+
+    rgb = page.photometric in (2, 6) and shape[-1] in (3, 4)
+    name = None
+    scale = None
+    colormap = None
+    contrast_limits = None
+    blending = None
+    channel_axis = None
+    visible = True
+
+    if page.photometric == 5:
+        # CMYK
+        channel_axis = axes.find("S")
+        if channel_axis >= 0 and shape[channel_axis] >= 4:
+            colormap = cmyk_colormaps()
+            name = ["Cyan", "Magenta", "Yellow", "Black"]
```

Comment:
This branch is `shape[channel_axis] >= 4` but you provide only 4 colormaps, 4 names etc? There should be some repeat?